catalog_tools/catalog.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+import functools
+
+import pandas as pd
+
+from catalog_tools.utils.binning import bin_to_precision
+
+REQUIRED_COLS = ['longitude', 'latitude', 'depth',
+                 'time', 'magnitude', 'magnitude_type']
```

Comment:
There might be situations where we don't really need the magnitude type, could it be optional?